backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java

```diff
@@ -427,15 +429,18 @@ protected String getFolderIdFromFoldersResponse(JsonNode response, String folder
      * @param fileName name of file to retrieve
      * @return string of file id
      */
-    protected String getFileIdFromFilesResponse(JsonNode response, String fileName) {
+    protected String getFileIdFromFilesResponse(JsonNode response, String fileName) throws CanvasAPIException {
         for (Iterator<JsonNode> it = response.elements(); it.hasNext(); ) {
             JsonNode folder = it.next();
             JsonNode name = folder.get("display_name");
             if (name != null && folder.get("display_name").asText().equalsIgnoreCase(fileName)) {
                 return folder.get("id").toString();
             }
         }
-        return null;
+        if (fileName.equals(SubmissionDirectoryService.MAKEFILE)) {
+            throw new MakefileNotFoundException();
+        }
+        throw new CanvasAPIException(fileName + " was not found.");
```

Comment:
Do we want to throw this exception every time this is called? This line doesn't have a condition for being thrown. 
Also, we may want to log these exceptions as well

Reply:
It only throws the exception if a matching file name isn't found. Before we were just returning null if we finished looping through the json, instead of throwing exceptions.

Reply:
they're both logged in fetchFileUnderCourseAssignmentFolder